ServiceBusMessageForwarder.IntegrationTests/TopicTests.cs

```diff
@@ -0,0 +1,393 @@
+            {
+                ignoreSubscriptions = subscriptionName;
+            });
+            "And a message is sent to the source topic".x(() =>
+            {
+                sourceClient.Send(new BrokeredMessage(_testMessage));
+            });
+            "When the service has run".x(() =>
+            {
+                new ServiceBusMessageForwarder(_logger, null, _sourceConnectionString, _destinationConnectionString, _ignoreQueues, ignoreTopics, ignoreSubscriptions).Run();
+            });
+            "Then the message is not forwarded to the destination topic".x(() =>
+            {
+                var destinationSubscriptionClient = SubscriptionClient.CreateFromConnectionString(_destinationConnectionString, topicName, subscriptionName);
+                var messages = destinationSubscriptionClient.PeekBatch(10);
+
+                messages.Count().Should().Be(0);
+            });
+            "And the message still exists in the source topic's subscription".x(() =>
+            {
+                var sourceSubscriptionClient = SubscriptionClient.CreateFromConnectionString(_sourceConnectionString, topicName, subscriptionName);
+                var messages = sourceSubscriptionClient.PeekBatch(10);
+                messages.Count().Should().Be(1);
+                messages.First().GetBody<Message>().Id.Should().Be(3011);
+            });
+
+            CleanupTopics(topicName);
+        }
+
+        [Scenario]
+        public void ForwardMessageWhenWhenNoSubscriptionsExist()
+        {
+            var topicName = $"_sbmf-{DateTime.UtcNow:yyyyMMddHHmmss}-{new Random().Next(10000, 99999)}";
+            var subscriptionName = "subscription1";
+            var ignoreTopics = "";
+            var ignoreSubscriptions = "";
+
+            var sourceClient = TopicClient.CreateFromConnectionString(_sourceConnectionString, topicName);
+
+            "Given a topic exists on the destination bus with no subscriptions".x(() =>
+            {
+                _destinationNamespaceManager.CreateTopic(topicName);
+
+                var subscriptions = _destinationNamespaceManager.GetSubscriptions(topicName);
+                foreach(var subscription in subscriptions)
+                    _destinationNamespaceManager.DeleteSubscription(topicName, subscription.Name);
+            });
+            "And the source topic has 1 subscription".x(() =>
+            {
+                _sourceNamespaceManager.CreateTopic(topicName);
+                _sourceNamespaceManager.CreateSubscription(topicName, subscriptionName);
+            });
+            "And a message is sent to the source topic".x(() =>
+            {
+                sourceClient.Send(new BrokeredMessage(_testMessage));
+            });
+            "When the service has run".x(() =>
+            {
+                new ServiceBusMessageForwarder(_logger, null, _sourceConnectionString, _destinationConnectionString, _ignoreQueues, ignoreTopics, ignoreSubscriptions).Run();
+            });
```

Comment:
If the message was forwarded, Should we check the message count at the destination?

Reply:
The message won't be retrievable in the destination in this instance, since there are no subscriptions (it is a concern of the destination, on whether it has 0 subscriptions, 1 subscription, multiple subscriptions, etc).
So we simply want to confirm that we did forward it from the source.